### DISCUSSION TITLE
Different render when in edit mode of IText

### DISCUSSION THREAD

**Discussion by markjerz** [fabricjs/fabric.js#7366] [Q&A]
Hi,

I am trying to subclass IText so that when you're in "edit mode" you can write normal text but that when you come out of edit mode the text is parsed and rendered in a different way.

I've naively overridden _render and added an if statement at the top along the lines of:

```
_render: function (ctx) {
  if (this.isEditing) {
      this.callSuper('_render', ctx);
      return;
  }

  .... parse text and render direct to canvas
```

It seems though, that when you come out of the edit mode you're not always guaranteed to have render called. 

Is there a better way of approaching this?

Thanks.

**Comment by ShaMan123** [comment]
I think you are looking for a complementary method.
Try [`enterEditing`](https://github.com/fabricjs/fabric.js/blob/2eabc92a3221dd628576b1bb029a5dc1156bdc06/src/mixins/itext_behavior.mixin.js#L331) and [`exitEditing`](https://github.com/fabricjs/fabric.js/blob/2eabc92a3221dd628576b1bb029a5dc1156bdc06/src/mixins/itext_behavior.mixin.js#L602)
And perhaps trigger another rendering cycle

**Reply by asturur** [reply]
i think this is correct.
i would expect the render happen anyway, but if it doesn't, events are they way to go.
